dom-fetch-client/src/main/scala/org/http4s/dom/FetchClient.scala

```diff
@@ -34,11 +34,11 @@ import scala.scalajs.js.typedarray.Uint8Array
 
 object FetchClient {
 
-  def apply[F[_]: Async]: Client[F] = Client[F] { req: Request[F] =>
-    (for {
-      body <- req.body.chunkAll.filter(_.nonEmpty).compile.last
-      response <- Async[F].fromPromise {
-        Async[F].delay {
+  def apply[F[_]](implicit F: Async[F]): Client[F] = Client[F] { req: Request[F] =>
+    for {
+      body <- req.body.chunkAll.filter(_.nonEmpty).compile.last.toResource
+      response <- F.fromPromise {
+        F.delay {
           val init = new RequestInit {}
```

Comment:
Should we create a general function to safely convert between a http4s Request and a RequestInit?

If `RequestInit` requires the body to be strict, then it'd have to be `Request[F] => Stream[F, RequestInit]` in the most general form. 

If this is the only place we use it, it may not be worth it.

Reply:
This is the only place we're using it so far, and I don't expect any others (fetch is the modern native client for browsers).